install.sh

```diff
@@ -0,0 +1,364 @@
+    rm -r $DIR &>/dev/null
+    RELEASE=${DIR%/}
+
+    # Getting rid of any old folders with the same name
+    rm -r $RELEASE-$version &>/dev/null
+
+    # Unzipping
+    unzip -q -u LanguageTool-stable.zip
+
+    mv $RELEASE "$RELEASE-$version"
+
+    # Cleaning up
+    rm LanguageTool-stable.zip
+}
+
+build () {
+    if [ -e languagetool ]; then
+        echo "Moved current languagetool directory to languagetool-previous"
+        mv languagetool languagetool-previous
+    fi
+
+    # Cloning from GitHub
+    git clone --depth "$clone_depth" https://github.com/languagetool-org/languagetool.git
+    cd languagetool || exit 2
+
+    # Seeing if maven is installed, and installing it if not
+    detect_maven
+
+    mvn clean test
+    if [ "$specifypackage" = YES ]; then
+        ./build.sh languagetool-$package package
+    else
+        ./build.sh languagetool-standalone package
+        ./build.sh languagetool-wikipedia package -DskipTests
+        ./build.sh languagetool-office-extension package -DskipTests
+    fi
+}
+
+build_quiet () {
+
+    if [ -e languagetool ]; then
+        echo "Moved current languagetool directory to languagetool-previous"
+        mv languagetool languagetool-previous
+    fi
+
+    # Cloning from GitHub
+    git clone -q --depth "$clone_depth" https://github.com/languagetool-org/languagetool.git
+    cd languagetool || exit 2
+
+    # Seeing if maven is installed, and installing it if not
+    detect_maven
+
+    mvn clean test
+    if [ "$specifypackage" = YES ]; then
+        ./build.sh languagetool-$package package
+    else
+        ./build.sh languagetool-standalone package
+        ./build.sh languagetool-wikipedia package -DskipTests
+        ./build.sh languagetool-office-extension package -DskipTests
+    fi
+}
+
+install_maven() {
+    echo "Installing maven . . ."
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        apt update -y
+        apt install maven -y
+
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        if ! [ -x "$(brew -v)" ]; then
+               /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+        fi
+        brew update
+        brew install maven
+
+    else
+            echo "Error: java is not installed and operating system detection error"
+            echo "   OS type not supported!"
+            echo "   Please install maven yourself or override automatic OS detection with -o <OS> See help for more details."
+    fi
+}
+
+install_java() {
+    echo "Java is not installed"
+    echo "Installing java . . ."
+
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        add-apt-repository ppa:webupd8team/java -y
+        apt update
+        apt install oracle-java8-installer -y
+
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        if ! [ -x "$(brew -v)" ]; then
+               /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```

Comment:
Retrieving a script from the Internet, may be contentious.Why not upload this script as well? 
Notice that homebrews with administrative permissions may cause the same concerns as PPAs.

I don't own a macOS, so macOS section needs to be reviewed by more people and can take some time.
Probably the best idea is to upload a base version concerning Linux (commenting out this else if) and after that, make another pull request with the macOS/darwin changes.